kmip/pie/objects.py

```diff
@@ -315,7 +331,7 @@ def __ne__(self, other):
             return NotImplemented
```

Comment:
`NotImplemented` is a special singleton value in Python that indicates that the comparison operator does not know how to compare itself to the given object. It is **not** equivalent to the `NotImplemented` exception.

In Python, comparison operators work in series. For example, say you want to compare objects `a` and `b`, so you do `a == b`. However, `a` doesn't know how to compare itself to `b`, so it returns `NotImplemented`. Python then, dynamically, runs `b == a`. If `b` knows how to compare itself to `a`, then the result of that comparison is used. If `b`, like `a`, doesn't know how to compare itself to `a`, Python will default to a raw object identity comparison, defaulting to a memory address comparison of the two objects.

Note the truth of `a == b` does not necessarily imply that `a != b` is false. Additionally, the truth of `a == b` does not imply that `b == a` is true. These comparison operators can be defined uniquely for every object to evaluate however you see fit.

For more information, see the docs on the [NotImplemented singleton](https://docs.python.org/2/library/constants.html#NotImplemented) and [comparison operators](https://docs.python.org/2/reference/datamodel.html#object.__eq__).